client/src/Features/MicrosoftLearn/MicrosoftLearnPage.tsx

```diff
@@ -1,78 +1,128 @@
-  header: [getCommonSpacingStyle(theme)]
-});
-export const MicrosoftLearnPage = styled(MicrosoftLearnPageInner, microsoftLearnPageStyles);
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License.
+ *--------------------------------------------------------------------------------------------*/
+
+import React, { useEffect, useState } from 'react';
+import { useObserver } from 'mobx-react-lite';
+import { useStore } from '../../Stores/Core';
+import { MicrosoftLearnList } from './MicrosoftLearnList';
+import { PageWrapper } from '../../Core/Components/Common/PageWrapper';
+import { themedClassNames } from '../../Core/Utils/FluentUI';
+import { SimpleComponentStyles, IStylesOnly, IThemeOnlyProps } from '../../Core/Utils/FluentUI/typings.fluent-ui';
+import { styled, Separator, mergeStyles } from '@fluentui/react';
+import { MicrosoftLearnSearch, MicrosoftLearnSearchStyles } from './MicrosoftLearnSearch';
+import {
+  MicrosoftLearnSelectedItemsList,
+  MicrosoftLearnSelectedItemsListStyles
+} from './MicrosoftLearnSelectedItemsList';
+import { getCommonSpacingStyle } from './MicrosoftLearnStyles';
+import { MicrosoftLearnFilterPane } from './MicrosoftLearnFilterPane';
+import { debounce } from 'lodash';
+import { MicrosoftLearnFilterTags } from './MicrosoftLearnFilterTags';
+import { useLocation } from 'react-router-dom';
+import { pagesDisplayNames } from '../../Router/Consts';
+
+type MicrosoftLearnPageStyles = SimpleComponentStyles<'root' | 'separator' | 'wrapper'>;
+
+export const TAB_SCREEN_SIZE = 768;
+
+const MicrosoftLearnPageInner = ({ styles }: IStylesOnly<MicrosoftLearnPageStyles>): JSX.Element => {
+  const learnStore = useStore('microsoftLearnStore');
+  const { filterStore, catalog } = learnStore
+  const location = useLocation();
+  const qsParams = new URLSearchParams(location.search);
+
+  useEffect(() => {
+    if (learnStore.catalog == null) {
+      learnStore.initializeCatalog(qsParams);
+    }
+  }, [learnStore, qsParams]);
+
+  // Filters are re-initialized based on the search params present in the location object (on browser back/forward), and the content gets updated accordingly.
+  // [Note]- This does not get called when filterStore.updateHistory is trigerred because we use H.createBrowserHistory to update the URL.
+  //       URL updation using it does not cause the current page to reload.
+
+  // TODO: Refactor the logic to start using useHistory hook for history management with proper UI update
+
+  useEffect(() => {
+    if (
+      catalog !== null &&
+      location.search !== '?' + filterStore.learnFilterUriParam && location.search !== filterStore.learnFilterUriParam
+    ) {
+      filterStore.initializeFilters(catalog!!, new URLSearchParams(location.search));
+    }  
+  }, [location]) // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Removing the hook and using a simple if statement. I added it back because the resize-hook was triggering it un-desirably. Given that has been moved to Filter Pane, we can have the if statement here as discussed before.